setup.py

```diff
@@ -118,32 +119,36 @@ def build_extension(self, ext):
         search = f"cpp/out/{preset}-build"
         candidates = glob.glob(search)
         if not candidates:
-            common_args = [
+            if preset == "*":
+                suffix = "-debug" if self.debug else "-release"
+                preset = ("windows-cl" if platform.system() == "Windows" else platform.system().lower()) + suffix
+            print(
+                f"Did not find build directory with '{search}'. Will configure and build using cmake preset {preset}",
+                file=sys.stderr,
+            )
+            _log_and_run(
+                cmake,
                 "-DTEST=NO",
                 f"-DBUILD_PYTHON_VERSION={sys.version_info[0]}.{sys.version_info[1]}",
-                *install_args,
-            ]
-            self._configure_cmake_using_preset(cmake, common_args, preset, search)
+                f"-DCMAKE_INSTALL_PREFIX={os.path.dirname(dest)}",
+                "--preset",
+                preset,
+                cwd="cpp",
+            )
             candidates = glob.glob(search)
 
         assert len(candidates) == 1, f"Specify {env_var} or use a single build directory. {search}={candidates}"
-        subprocess.check_call([cmake, "--build", candidates[0], "--target", ext.name])
-        subprocess.check_call(
-            [cmake, *install_args, "-DCOMPONENT=Python_Lib", "-P", candidates[0] + "/cmake_install.cmake"]
-        )
+        try:
+            # Python API is not cgroups-aware yet, so use CMake:
```

Comment:
What environment is affected by this?

Reply:
Comes to think of it, none that we use at the moment...

The CMake script was salvaged from the CMakeLists.txt and also used to set the concurrency for tests. It also does other checks, so I am not eager to rewrite in Python at the moment.